src/main/java/core/basesyntax/strategy/PurchaseOperationHandlerImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandlerImpl implements OperationHandler {
+    @Override
+    public void handleOperation(FruitTransaction transaction) {
+        String fruitName = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+
+        try {
+            if (FruitStorage.getFruit(fruitName).isPresent()) {
+                FruitTransaction fruit = FruitStorage.getFruit(fruitName).get();
+                int updatedQuantity = fruit.getQuantity() - quantity;
+                if (updatedQuantity < 0) {
+                    throw new RuntimeException("Invalid fruit "
+                            + "quantity after purchase: " + fruitName);
+                }
+                fruit.setQuantity(updatedQuantity);
+            }
+        } catch (Exception e) {
+            System.out.println(e);
+        }
```

Comment:
do you need try- catches in you handlers? what exceptions you expect there?

Reply:
No need try-catche, not will exception